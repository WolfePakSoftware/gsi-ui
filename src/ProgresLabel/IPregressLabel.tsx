export interface IProgressLabelProps {
    total: number,
    width?: string,
    minWidth?: string,
    segments: ISegment[]
}

export interface ISegment {
    key: string,
    type: 'first' | 'center' | 'last' | 'unique',
    label: string,
    amount: number,
    color?: string
}