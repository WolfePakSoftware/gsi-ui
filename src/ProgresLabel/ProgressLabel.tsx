import React, {Component} from 'react';
import {
    dividerProgresVertical,
    ProgressLabelSegment, ProgressLabelSegmentCenter,
    ProgressLabelSegmentFirst,
    ProgressLabelSegmentLast, ProgressLabelSegmentUnic,
    SpanText
} from "./style";
import {Divider} from "@blueprintjs/core";
import {IProgressLabelProps, ISegment} from "./IPregressLabel";

class ProgressLabel extends Component<IProgressLabelProps> {
    constructor(props: IProgressLabelProps) {
        super(props);
    }

    createSegment(total: number, segment: ISegment) {
        let segmentResult = (<div>default</div>);
        const percent = (segment.amount * 100) / total;
        switch (segment.type) {
            case 'first':
                segmentResult =
                    <ProgressLabelSegmentFirst key={segment.key} Width={`${percent}%`}>
                        <SpanText>
                            {`${segment.amount} ${segment.label}`}
                        </SpanText>
                    </ProgressLabelSegmentFirst>;
                break;
            case 'last':
                segmentResult =
                    <ProgressLabelSegmentLast key={segment.key} Width={`${percent}%`}>
                        <SpanText>
                            {`${segment.amount} ${segment.label}`}
                        </SpanText>
                    </ProgressLabelSegmentLast>;
                break;
            case 'unique':
                segmentResult =
                    <ProgressLabelSegmentUnic key={segment.key} Width={`${percent}%`}>
                        <SpanText>
                            {`${segment.amount} ${segment.label}`}
                        </SpanText>
                    </ProgressLabelSegmentUnic>;
                break;
            case 'center':
                segmentResult =
                    <ProgressLabelSegmentCenter key={segment.key} Width={`${percent}%`}>
                        <SpanText>
                            {`${segment.amount} ${segment.label}`}
                        </SpanText>
                    </ProgressLabelSegmentCenter>;
                break
        }
        return segmentResult;
    }

    render() {
        const {total, segments} = this.props;
        const segmentsRender = segments.map(segment => {
            return this.createSegment(total, segment);
        });
        return (
            <ProgressLabelSegment>
                {segmentsRender}
            </ProgressLabelSegment>
        );
    }
}

export default ProgressLabel;