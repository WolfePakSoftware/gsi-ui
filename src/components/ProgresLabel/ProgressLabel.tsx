import React, { Component } from 'react';
import {
  ProgressLabelSegment,
  ProgressLabelSegmentCenter,
  ProgressLabelSegmentFirst,
  ProgressLabelSegmentLast,
  ProgressLabelSegmentUnic,
  SpanText
} from './style';
import { IProgressLabelProps, ISegment } from './IPregressLabel';

export class ProgressLabel extends Component<IProgressLabelProps> {
  constructor(props: IProgressLabelProps) {
    super(props);
  }

  createSegment(total: number, segment: ISegment) {
    let segmentResult = <div>default</div>;
    const percent = (segment.amount * 100) / total;
    switch (segment.type) {
      case 'first':
        segmentResult = (
          <ProgressLabelSegmentFirst key={segment.key} Width={`${percent}%`}>
            <SpanText>{`${segment.amount} ${segment.label}`}</SpanText>
          </ProgressLabelSegmentFirst>
        );
        break;
      case 'last':
        segmentResult = (
          <ProgressLabelSegmentLast key={segment.key} Width={`${percent}%`}>
            <SpanText>{`${segment.amount} ${segment.label}`}</SpanText>
          </ProgressLabelSegmentLast>
        );
        break;
      case 'unique':
        segmentResult = (
          <ProgressLabelSegmentUnic key={segment.key} Width={`${percent}%`}>
            <SpanText>{`${segment.amount} ${segment.label}`}</SpanText>
          </ProgressLabelSegmentUnic>
        );
        break;
      case 'center':
        segmentResult = (
          <ProgressLabelSegmentCenter key={segment.key} Width={`${percent}%`}>
            <SpanText>{`${segment.amount} ${segment.label}`}</SpanText>
          </ProgressLabelSegmentCenter>
        );
        break;
    }
    return segmentResult;
  }

  render() {
    const { total, segments } = this.props;
    const segmentsRender = segments.map(segment => {
      return this.createSegment(total, segment);
    });
    return <ProgressLabelSegment>{segmentsRender}</ProgressLabelSegment>;
  }
}
