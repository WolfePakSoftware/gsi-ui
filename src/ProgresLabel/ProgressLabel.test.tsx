import React from "react";
import ReactDOM from "react-dom";
import {ProgressLabel} from "./ProgressLabel";
import {ISegment} from "./IPregressLabel";
import renderer from 'react-test-renderer';

it("renders without crashing Progress Label", () => {
    const div = document.createElement("div");
    const segments: ISegment[] = [{
        key: 'first',
        type: 'first',
        label: 'Hola Mundo',
        amount: 10
    }, {
        key: 'last',
        type: 'last',
        label: 'Soy yo',
        amount: 5
    }];
    ReactDOM.render(<ProgressLabel total={15} segments={segments}/>, div);
    ReactDOM.unmountComponentAtNode(div);
});
it('render correctly ProgressLabel component one segment', () => {
    const segments: ISegment[] = [{
        key: 'unique',
        type: 'unique',
        label: 'Hola Mundo',
        amount: 10
    }];
    const TextInputComponent = renderer.create(<ProgressLabel total={10} segments={segments}/>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

it('render correctly ProgressLabel component two segments', () => {
    const segments: ISegment[] = [{
        key: 'first',
        type: 'first',
        label: 'Hola Mundo',
        amount: 10
    }, {
        key: 'last',
        type: 'last',
        label: 'Soy yo',
        amount: 5
    }];
    const TextInputComponent = renderer.create(<ProgressLabel total={15} segments={segments}/>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

it('render correctly ProgressLabel component three segment', () => {
    const segments: ISegment[] = [{
        key: 'first',
        type: 'first',
        label: 'Hola Mundo',
        amount: 8
    },
        {
            key: 'center',
            type: 'center',
            label: 'Soy yo',
            color: 'red',
            amount: 4
        }, {
            key: 'last',
            type: 'last',
            label: 'Soy yo',
            amount: 3
        }];
    const TextInputComponent = renderer.create(<ProgressLabel total={15} segments={segments}/>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});
