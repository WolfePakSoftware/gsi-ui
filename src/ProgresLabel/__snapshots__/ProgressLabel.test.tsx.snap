// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly ProgressLabel component one segment 1`] = `
<div
  className="sc-bdVaJa jYmnxs"
>
  <div
    className="sc-ifAKCX eDCHnM"
  >
    <span
      className="sc-EHOje YINzU"
    >
      10 Hola Mundo
    </span>
  </div>
</div>
`;

exports[`render correctly ProgressLabel component three segment 1`] = `
<div
  className="sc-bdVaJa jYmnxs"
>
  <div
    className="sc-bwzfXH EarXg"
  >
    <span
      className="sc-EHOje YINzU"
    >
      8 Hola Mundo
    </span>
  </div>
  <div
    className="sc-bxivhb glTCqr"
  >
    <span
      className="sc-EHOje YINzU"
    >
      4 Soy yo
    </span>
  </div>
  <div
    className="sc-htpNat naIiR"
  >
    <span
      className="sc-EHOje YINzU"
    >
      3 Soy yo
    </span>
  </div>
</div>
`;

exports[`render correctly ProgressLabel component two segments 1`] = `
<div
  className="sc-bdVaJa jYmnxs"
>
  <div
    className="sc-bwzfXH xTpsF"
  >
    <span
      className="sc-EHOje YINzU"
    >
      10 Hola Mundo
    </span>
  </div>
  <div
    className="sc-htpNat kBBuKK"
  >
    <span
      className="sc-EHOje YINzU"
    >
      5 Soy yo
    </span>
  </div>
</div>
`;
